### ISSUE TITLE
Pycuda interoperability

### ISSUE THREAD

**Issue by themightyoarfish** [pytorch/pytorch#9624]
For doing computations on tensors with `pycuda`, I can do this
So going from `Tensor` to `pycuda.gpuarray.GPUArray` 
```
def tensor_to_gpuarray(tensor):
    if not tensor.is_cuda:
         raise ValueError('Cannot convert CPU tensor to GPUArray (call `cuda()` on it)')
     else:
         return GPUArray(tensor.shape, dtype=torch_dtype_to_numpy(tensor.dtype), gpudata=tensor.data_ptr())
```
however, I'm stumped on how to convert back to a Tensor without copying the data. How can this be done?

**Comment by themightyoarfish** [comment]
Yes, that seems to be the only viable solution right now. But if we assume that I want to have some complex chain of calculations where calculating the final shape would we cumbersome, it would be nice to just write down the formulas with pycuda or scikit-cuda and then wrap the resulting memory/GPUArray back into a tensor.

**Comment by SsnL** [comment]
This is probably hard because pycuda doesn't support dlpack..

**Comment by mruberry** [comment]
Related: https://github.com/inducer/pycuda/issues/226. As @SsnL mentions that's probably a blocking issue.

**Comment by leofang** [comment]
I bet most workloads can be quickly converted using CuPy's functionalities, so I would not stick to PyCUDA if interoperating with other libraries is needed.
